error.go

```diff
@@ -25,6 +28,47 @@ func (e *APIError) Error() string {
 	return e.Message
 }
 
+func (e *APIError) UnmarshalJSON(data []byte) (err error) {
+	var rawMap map[string]json.RawMessage
+	err = json.Unmarshal(data, &rawMap)
+	if err != nil {
+		return
```

Comment:
Maybe the simplest way to cover this would be to pass some invalid data. We also have testable jsonMarshaller [here](https://github.com/sashabaranov/go-openai/blob/master/request_builder_test.go#L22), not sure if it would be possible to accommodate interface-based jsonUnmarshaller here.

Reply:
I added a test for this block, but I could not pass the invalid data through `json.Unmarshal` because the standard library [filters out malformed data](https://cs.opensource.google/go/go/+/refs/tags/go1.20.3:src/encoding/json/decode.go;l=102) before sending it to the unmarshaller. I ended up just calling the method directly in the test.